django/db/models/expressions.py

```diff
@@ -2140,3 +2140,72 @@ class ValueRange(WindowFrame):
 
     def window_frame_start_end(self, connection, start, end):
         return connection.ops.window_frame_range_start_end(start, end)
+
+
+class RowTuple(ExpressionList):
+    template = "(%(expressions)s)"
+
+    def as_mysql(self, compiler, connection, **extra_context):
+        # MySQL requires the ROW() function to be used for tuples in VALUES clauses.
+        if not connection.mysql_is_mariadb:
+            extra_context["template"] = "ROW(%(expressions)s)"
+        return self.as_sql(compiler, connection, **extra_context)
+
+    def __str__(self):
+        values = self.arg_joiner.join(str(arg) for arg in self.source_expressions)
+        return self.template % {"expressions": values}
+
+    def __getitem__(self, item):
+        return self.source_expressions[item]
+
+
+class RowTupleValues(ExpressionList):
+    template = "VALUES %(expressions)s"
+
+    def __init__(self, expression_list, pk_field, field_list, **extra):
+        expressions = (
+            (
+                RowTuple(*expressions)
+                if not isinstance(expressions, RowTuple)
+                else expressions
+            )
+            for expressions in expression_list
+        )
+        self.pk_field = pk_field
+        self.field_list = field_list
+        super().__init__(*expressions, **extra)
+
```

Comment:
This could be done in a follow up but I suspect the `SELECT * FROM unnest` strategy could be used on Postgres as well here as well #18847 as we pre-emptively make sure that all values are literals.